agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -304,6 +304,9 @@ public static class PreviewConfiguration {
     // telemetry (they are much smaller so a larger queue size is ok)
     public int metricsExportQueueCapacity = 65536;
 
+    // disk persistence has a default capacity of 50MB
+    public int diskPersistenceMaxSizeMb = 50;
```

Comment:
2.x SDK enforces a limit of 1 GB:

https://github.com/microsoft/ApplicationInsights-Java/blob/d3a73be1531a258a3627f36542cc36050464fd3b/core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionFileSystemOutput.java#L83

I'm not really sure the reason, can you see if Application Insights .NET SDK has a max capacity that users can configure? if they do, we may want to implement that max too

also, I wonder if we should allow `0`, meaning disable disk persistency, which users on read-only filesystems could use (and then we wouldn't need to log warning for those users)

Reply:
dotnet is default to 50MB as well. they use 30 seconds as the default sending interval. same as us.. it's not configured. we could ignore value that is less than 50? and document it?

Reply:
dotnet [source code](https://github.com/microsoft/ApplicationInsights-dotnet/blob/405fd6a9916956f2233520c8ab66110a1f9dcfbc/BASE/src/Microsoft.ApplicationInsights/Channel/InMemoryTransmitter.cs#:~:text=private%20TimeSpan%20sendingInterval%20%3D%20TimeSpan.FromSeconds(30)%3B)

Reply:
> dotnet is default to 50MB as well

do they allow user to configure this? and if so, do they have a maximum value that the user can set?

Reply:
it's fixed. no configuration.

Reply:
that makes me a little concerned that there might be a reason behind why .NET SDK doesn't support this (since they typically have so many more users and configuration knobs in general)

@TimothyMothra do you happen to know if there's any specific reason .NET SDK doesn't support configuration of the disk size / do you know if the topic has come up before? thx

Reply:
.NET SDK's disk storage capacity is configurable: [SeverTelemetryChannel](https://github.com/microsoft/ApplicationInsights-dotnet/blob/405fd6a9916956f2233520c8ab66110a1f9dcfbc/BASE/src/ServerTelemetryChannel/ServerTelemetryChannel.cs#L188-L196)

Reply:
Thx, does the .Net SDK impose any limit on how large the user can configure it to be?

Reply:
but wait, is it exposed to customers though? is it available in the public interface?

Reply:
> Thx, does the .Net SDK impose any limit on how large the user can configure it to be?

Only limit is the maximum value of `long` itself.

Reply:
looks like it's public API. based on the code, it can be any number.

Reply:
awesome, thx all 👍

Reply:
https://github.com/microsoft/Telemetry-Collection-Spec/pull/83